### PR TITLE
[noTicket][risk=low]Send Workspace Creator information to RDR

### PR DESCRIPTION
**Pull request by NehaBroad**
Send workspace creator information: id and name to rdr as Researcher hub need to display workspace owner information along with workspace details.

**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java [hunk 1]
```diff
@@ -447,6 +448,22 @@ public void updateDbRdrExport(RdrEntity entity, List<Long> idList) {
...
   }
 
+  /**
+   * Creates RdrWorkspaceCreator object that contains workspace creator information like id, given
+   * name and lastName
+   *
+   * @param creatorUser
+   * @return RdrWorkspaceCreator object
+   */
+  RdrWorkspaceCreator toRdrWorkspaceCreator(DbUser creatorUser) {
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java:458]: Please make a new MapStruct mapper for this. There's a guide and tutorial in the top level README.

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java [hunk 2]
```diff
@@ -71,7 +72,7 @@
 
   private DbUser dbUserWithEmail;
   private DbUser dbUserWithoutEmail;
-  private DbWorkspace mockWorkspace, mockDeletedWorkspace;
+  private DbWorkspace mockWorkspace, mockDeletedWorkspace, mockCreatorWorkspace;
```
Comment (jaycarlton) [inline api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:75]: nit: Usually I'd use `@Mock` for mocking value objects (and `@MockBean` for beans). There may be a reason not to do so here.

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java [hunk 3]
```diff
@@ -117,9 +118,14 @@ public void setUp() {
 
     mockDeletedWorkspace =
         buildDbWorkspace(2, "workspace_del", "workspaceNs", WorkspaceActiveStatus.DELETED);
-
+    mockDeletedWorkspace.setCreator(dbUserWithEmail);
     when(mockWorkspaceDao.findDbWorkspaceByWorkspaceId(2)).thenReturn(mockDeletedWorkspace);
 
+    mockCreatorWorkspace =
+        buildDbWorkspace(3, "mock_workspace_name", "workspaceNS", WorkspaceActiveStatus.ACTIVE);
+    mockCreatorWorkspace.setCreator(dbUserWithoutEmail);
```
Comment (jaycarlton) [inline api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:126]: I would have expected the creator to be a field passed in to `buildDbWorkspace`.

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java [hunk 4]
```diff
@@ -215,6 +221,25 @@ public void exportWorkspace_FocusOnUnderservedPopulation() throws ApiException {
     verify(mockRdrApi).exportWorkspaces(Arrays.asList(rdrWorkspace));
   }
 
+  @Test
+  public void exportWorkspace_CreatorInformation() throws ApiException {
+    List<Long> workspaceID = new ArrayList<>();
```
Comment (jaycarlton) [inline api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:226]: `workspaceIds` plural.

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java [hunk 5]
```diff
@@ -215,6 +221,25 @@ public void exportWorkspace_FocusOnUnderservedPopulation() throws ApiException {
...
+    workspaceID.add(1l);
+    rdrExportService.exportWorkspaces(workspaceID);
+    verify(rdrExportDao, times(1)).save(anyList());
```
Comment (jaycarlton) [inline api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:229]: Is there anything else you can verify?

Comment (jaycarlton) [reply api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:229]: If you're using a real DAO, then you should be able to fetch the saved object and compare fields.

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java [hunk 6]
```diff
@@ -215,6 +221,25 @@ public void exportWorkspace_FocusOnUnderservedPopulation() throws ApiException {
...
+
+    RdrWorkspace rdrWorkspace = toDefaultRdrWorkspace(mockWorkspace);
+    verify(mockRdrApi).exportWorkspaces(Arrays.asList(rdrWorkspace));
+
+    workspaceID = new ArrayList<>();
+    workspaceID.add(3l);
+    rdrExportService.exportWorkspaces(workspaceID);
+    rdrWorkspace = toDefaultRdrWorkspace(mockCreatorWorkspace);
+    rdrWorkspace.setCreator(
+        new RdrWorkspaceCreator().userId(2l).familyName("email").givenName("icannothas"));
```
Comment (jaycarlton) [inline api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:239]: With fluent constructsors or builders, please break before every new parameter.

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java (hunk 5 above)
Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:229]: i actually do not need to verify this, the purpose of this test is to ensure the creator information is different

Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:229]: and i am verifying the RdrWorkspace converted object attribute value is matched via
    verify(mockRdrApi).exportWorkspaces(Arrays.asList(rdrWorkspace));

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java (hunk 6 above)
Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:256]: so spotlessApply is putting them in one line. i had it like ....
            .userId(2l)
            .familyName("email")
            .givenName("icannothas"));

File: api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java [hunk 7]
```diff
@@ -274,45 +272,9 @@ private RdrResearcher toRdrResearcher(DbUser dbUser) {
...
-    rdrWorkspace.setModifiedTime(
-        dbWorkspace.getLastModifiedTime().toLocalDateTime().atOffset(offset));
-
-    RdrWorkspace.StatusEnum workspaceRDRStatus =
-        dbWorkspace.getWorkspaceActiveStatusEnum() == WorkspaceActiveStatus.ACTIVE
-            ? RdrWorkspace.StatusEnum.ACTIVE
-            : RdrWorkspace.StatusEnum.INACTIVE;
-
-    rdrWorkspace.setStatus(workspaceRDRStatus);
+    RdrWorkspace rdrWorkspace = rdrMapper.toRdrModel(dbWorkspace);
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java:275]: Yay! look how small it is!

File: api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.model.SpecificPopulationEnum;
+import org.pmiops.workbench.model.WorkspaceActiveStatus;
+import org.pmiops.workbench.rdr.model.RdrWorkspace;
+import org.pmiops.workbench.rdr.model.RdrWorkspaceDemographic;
+import org.pmiops.workbench.utils.mappers.MapStructConfig;
+
+@Mapper(config = MapStructConfig.class)
+public interface RdrMapper {
+  @Mapping(target = "modifiedTime", source = "lastModifiedTime")
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java:21]: This is great, but even better would be to have the fields named the same way.

Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java:21]: It's better if we can order them by `source, target`, and then use a string utility to alphabetize them.

File: api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+  @Mapping(target = "intendToStudy", source = "intendedStudy")
+  @Mapping(target = "diseaseFocusedResearchName", source = "diseaseOfFocus")
+  @Mapping(target = "ethicalLegalSocialImplications", source = "ethics")
+  @Mapping(target = "scientificApproaches", source = "scientificApproach")
+  @Mapping(target = "findingsFromStudy", source = "anticipatedFindings")
+  @Mapping(target = "status", source = "workspaceActiveStatusEnum")
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java:27]: We've already got different names for the db and model classes here, and now the RDR model is yet another mapping. This is just way more complicated than it needs to be. But the mapper sweeps it all under the rug in a really nice way.

File: api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+  @Mapping(target = "workspaceUsers", ignore = true)
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java:28]: For anything you need to ignore, please put a comment indicating why it's required/safe/desired.

File: api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+  default boolean toModelFocusOnUnderrepresentedPopulation(
+      Set<SpecificPopulationEnum> dbSpecificPopulationSet) {
+    return dbSpecificPopulationSet != null && dbSpecificPopulationSet.size() > 0;
+  }
+
+  default RdrWorkspaceDemographic toModelWorkspaceDemographic(
+      Set<SpecificPopulationEnum> dbPopulationEnumSet) {
+    RdrWorkspaceDemographic rdrDemographic = new RdrWorkspaceDemographic();
+
+    rdrDemographic.setAccessToCare(
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java:55]: nit: you can get around all ternary operators if you just set all the flages to unset first, then iterate over the enums and use a mapping function, sort of like the `PropertyMapper` class. Probably too fancy to bother with, but it's a neat trick.

So it'd be like `AccessToCare.setRdrFlag(rdrDemogrphic);` You can just zip down the set and call the `setRdrFlag()` method on each one for the thing and avoid all the branching.

File: api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+            : RdrWorkspaceDemographic.GenderIdentityEnum.UNSET);
+
+    rdrDemographic.setSexAtBirth(
+        dbPopulationEnumSet.contains(SpecificPopulationEnum.SEX)
+            ? RdrWorkspaceDemographic.SexAtBirthEnum.INTERSEX
+            : RdrWorkspaceDemographic.SexAtBirthEnum.UNSET);
+
+    rdrDemographic.setRaceEthnicity(
+        dbPopulationEnumSet.stream()
+            .map(RdrExportEnums::specificPopulationToRaceEthnicity)
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrMapper.java:97]: You may want to find out when this returns null. Maybe a unit test there. In general, if you have to do a filter on non-null, there might be a constraint you could add upstream so this doesn't happen. Not always though.

File: api/src/main/resources/rdr.yaml [hunk 13]
```diff
@@ -237,6 +240,22 @@ definitions:
...
+  RdrWorkspaceCreator:
+    type: object
+    description: Details for workspace creator
+    required:
+      - userId
+      - givenName
+      - familyName
+    properties:
+      userId:
+        type: integer
```
Comment (jaycarlton) [inline api/src/main/resources/rdr.yaml:252]: please add a description about this ID. Like is it for our database or some other thing?

File: api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java (hunk 6 above)
Comment (jaycarlton) [reply api/src/test/java/org/pmiops/workbench/rdr/RdrExportServiceImplTest.java:256]: ok. Spotless isn't the best poet.

**Pull request merged**